aten/src/ATen/native/cudnn/RNN.cpp

```diff
@@ -700,21 +822,27 @@ namespace cudnn_rnn {
     std::vector<Tensor> params_arr;
     size_t params_stride0;
     std::tie(params_arr, params_stride0) = get_parameters(handle, rnn, rnn_desc, x_desc, w_desc, weight_buf, include_bias);
-
     MatrixRef<Tensor> weight{weight_arr, static_cast<size_t>(weight_stride0)},
                       params{params_arr, params_stride0};
 
     // Copy weights
     _viewOrCopyParams(weight, params, /*copy=*/true, allow_type_change);
-
     if (set_orig_weights_to_flat_buf) {
       // Update the storage
       for (size_t i = 0; i < weight.size(0); i++) {
-        for (auto orig_param_it = weight[i].begin(), new_param_it = params[i].begin();
-             orig_param_it != weight[i].end() && new_param_it != params[i].end();
-             orig_param_it++, new_param_it++) {
-          auto orig_param = *orig_param_it, new_param = *new_param_it;
-          orig_param.set_(new_param.view_as(orig_param));
+        // There is a special case for LSTM with projections and no bias,
+        // where weight copy is done in 0->0, 1->1, 2->4 layout
+        if (weight[i].size() == 3) {
```

Comment:
wouldn't weight[i].size() be 3 also for GRU with no bias?

Reply:
Added check for params.size as well to distinguish these cases